flipperdevices/flipperzero-firmware
# FurEventLoop: add support for FuriEventFlag, simplify API

# What's new

- FuriEventLoop: 
  - FuriEventFlag integration
  - Abolish callback return value (Level events are now always rescheduled)
  - Update docs
- Unit tests: 
  - (NEW) FuriEventFlag, FuriSemaphore, FuriStreamBuffer
  - Refactoring
  - Add stub tests for Furi OS-level primitives
- Examples: FuriEventFlag example

# Verification 
1. Run unit tests (everything should be OK)
2. Run the new example application: 
   ```sh
   ./fbt launch APPSRC=example_event_loop_event_flags
   ```
   Press some buttons and observe the debug output.
3. Use the firmware normally, no regressions should be present.

# Checklist (For Reviewer)
